src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs

```diff
@@ -252,7 +252,64 @@ public bool TryGet(ref Position cursor, out ReadOnlyMemory<byte> data, bool adva
 
             return result;
         }
-        
+
+        public int Seek(out Position result, byte byte0)
```

Comment:
Why does it return an int? Isn't the int always equal to byte0 or -1? If yes, the API should be:
```c#
 public bool TrySeek(out Position position, byte value);
// or
public Position? Seek(byte value);
```

Also, since we call integer index returning method IndesOf, why not call these PositionOf?

Reply:
It works for single byte overload, what about multi-byte ones? We returned int to be able to quickly know what byte was found, without slicing or creating the reader.

Reply:
The only overload that you are actually using in the pipelines/parser the one that takes a single value.

The other overloads should be called PositionOfAny (to be consistent with IIndexOfAny). These are not used by the http parser, and so maybe we don't have to return the found value, just like we don't return it from IndexOfAny. Alternativelly, we should redesign IndexOfAny (in MemoryExtensions)

Reply:
I like changing it to be more like `IndexOfAny`, but not returning byte that was found would make things harder because you need to slice just to get it.

Reply:
I'll change it to `PositionOf[Any]`

Reply:
Why do we need the [Any] versions? Nothing uses them except for tests.

Reply:
If we go with, "cut everything we don't use" - no. But then we can cut all of them and implement it in kestrel.

Reply:
Sorry, I was not clear. I was asking whether we need the super optimized versions that return the byte found.
 
What about the following:
```c#
public Position? PositionOf(byte value);
public Position? PositionOf(ReadOnlySpan<byte> value);
public Position? PositionOfAny(ReadOnlySpan<byte> values);
```

Reply:
Often the first things you do after you found the position is to check what character is it. We did it for URL parsing, looked for `'?', ' '` and depending what was found executed different logic.